dashboard/src/app/factories/load-factory/load-factory.controller.ts

```diff
@@ -125,14 +125,30 @@ export class LoadFactoryController {
           this.getLoadingSteps()[this.getCurrentProgressStep()].hasError = true;
           this.getLoadingSteps()[this.getCurrentProgressStep()].logs = 'Factory has no workspace config.';
         } else {
-          this.fetchWorkspaces();
+          this.loadFactoryService.goToNextStep();
+          this.$timeout(() => {
+             this.processFactorySource();
+          }, 1500);
```

Comment:
it looks very fragile to have a timeout, no ?

Reply:
No, that makes step to show smoothly, instead of just flashing in 0.001 s

Reply:
shouldn't it be adding a new step instead of adding text to an existing step ?